### PR TITLE
Use cloudant client

### PR DESCRIPTION
**Pull request by smithsz**
Thanks for your hard work, please ensure all items are complete before opening.

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](https://github.com/cloudant/nodejs-cloudant/blob/master/DCO1.1.txt)
- [x] You have added tests for any code changes
- [x] You have updated the [CHANGES.md](https://github.com/cloudant/nodejs-cloudant/blob/master/CHANGES.md)
- [x] You have completed the PR template below:

## What
Use the new Cloudant client in `cloudant.js`.

## How
Pass the Cloudant client instance through to the underlying nano object.

## Testing
Includes additional/updated unit tests.

### CODE REVIEW

File: test/legacy/plugin.js [hunk 1]
```diff
@@ -154,28 +158,6 @@ describe('promise plugin', function() {
   });
 });
 
-describe('custom plugin', function() {
```
Comment (smithsz) [inline test/legacy/plugin.js]: This is now an invalid test. All custom plugins must be a class that inherits from `BasePlugin`.

File: test/legacy/api.js [hunk 2]
```diff
@@ -574,23 +568,12 @@ describe('Changes follower', function() {
     var mocks = nock(SERVER)
       .get('/' + dbName)
       .reply(200, { update_seq: '2-g1AAAADbeJzLYWBgYMlgTmGQTUlKzi9KdUhJMtUrzsnMS9dLzskvTUnMK9HLSy3JASpjSmRIsv___39WIgORGpIcgGRSPVgPI5F68liAJEMDkAJq20-8XRB9ByD6QPZlAQCMOkh4', db_name: dbName, sizes: { file: 58038, external: 8, active: 2166 }, purge_seq: 0, other: { data_size: 8 }, doc_del_count: 0, doc_count: 2, disk_size: 58038, disk_format_version: 6, compact_running: false, instance_start_time: '0' })
-      .post('/' + dbName)
-      .reply(200, {ok: true, id: docId, rev: '2-a1933e6ba0b5ac9868cd6f5adc12dc5f'})
       .filteringPath(/[?&](since=.*|feed=continuous|heartbeat=.*)/g, '') // Strip out the standard parameters: ?since &feed, &heartbeat
       .get('/' + dbName + '/_changes')
       .reply(200, '{"seq":"3-g1AAAAEzeJzLYWBgYMlgTmGQTUlKzi9KdUhJMtcrzsnMS9dLzskvTUnMK9HLSy3JASpjSmRIsv___39WBpCVCxRgT05OSjQ0SyNSe5IDkEyqB5vAnMgENsHM1CDZPDGFkH4ibchjAZIMDUAKaMl-hDtTLZMMktOMSDLlAMQUsG8ZIaYYWpqmGCVnAQDI71_n","id":"' + docId + '","changes":[{"rev":"2-a1933e6ba0b5ac9868cd6f5adc12dc5f"}]}\n');
 
-    // Make an update that will trigger the "since=now" follower.
```
Comment (smithsz) [inline test/legacy/api.js]: This test runs as a mock only. Therefore the flakey update_doc after 1sec logic is redundant.

File: cloudant.js [hunk 3]
```diff
@@ -233,55 +209,11 @@ function Cloudant(options, callback) {
...
-      if (er) {
-        callback(er);
-      } else {
-        if (cookie) {
-          requestDefaults.headers.cookie = cookie;
-        }
-
-        callback(null, nano, pong);
-      }
+    nano.cc.addPlugins('cookieauth');
```
Comment (ricellis) [inline cloudant.js]: So if I'm understanding this correctly, if you request the ping callback then the `cookieauth` plugin is enabled by default? I thought we were going to turn `cookieauth` on by default unless a plugin list was supplied that explicitly removed it?

File: test/legacy/plugin.js (hunk 1 above)
Comment (ricellis) [reply test/legacy/plugin.js]: As discussed I think we should deprecate the old style custom plugin a bit more gently. I'm happy for that second path to be done in another PR if it is easier.

File: cloudant.js (hunk 3 above)
Comment (smithsz) [reply cloudant.js]: Yeah, the ping callback enables cookie auth. I don't like it much. But it's following the old pattern.

I haven't turned cookie auth on by default yet. It's something we can do. I wasn't too sure how a client would go about disabling it though. The cloudant client is exposed so I could add a new `removePlugins` method (to mirror `addPlugins`), i.e.
```js
const client = Cloudant({ plugins: ['retry'] });
client.cc.removePlugins('cookieauth');
```
I don't want to clutter this pr even more so I'll open a new one for this.

File: test/legacy/plugin.js (hunk 1 above)
Comment (smithsz) [reply test/legacy/plugin.js]: Okay, this makes sense. I'll put together a PR to reinstate this test and handle legacy plugins.

**Pull request merged**